### DISCUSSION TITLE
How is Vanara compared to Vortice.Win32?

### DISCUSSION THREAD

**Discussion by ghost** [dahall/Vanara#431] [Q&A] [answered]
https://github.com/amerkoleci/Vortice.Win32

Raised this question after I found #364 and #367.

**Comment by dahall** [comment, marked as answer]
It looks like Vortice is very similar to the PInvoke project. It is a generator based on a logical extraction of Win32 functions and constants. It will likely generate just those raw functions using `unsafe` method types. Vanara is a wrapper that exposes those same functions and constants but in a complete .NET manner with supporting types, memory management, marshalers, etc.
